### PR TITLE
Emote reply

### PR DESCRIPTION
**Pull request by 4ndriH**
I have no clue if that works tbh ^^

### CODE REVIEW

File: ETHDINFKBot/Handlers/MessageHandler.cs [hunk 1]
```diff
@@ -297,7 +299,14 @@ private async void EmojiDetection()
 
                                 using (var stream = new MemoryStream(File.ReadAllBytes(emote.LocalPath)))
                                 {
-                                    await SocketMessage.Channel.SendFileAsync(stream, $"{emote.EmoteName}.gif");
+                                    if (Context.Message.ReferencedMessage != null)
+                                    {
+                                        await Context.Channel.SendFileAsync(stream, $"{emote.EmoteName}.gif", "", false, null, null, false, null, new Discord.MessageReference(Context.Message.ReferencedMessage.Id));
```
Comment (BattleRush) [inline ETHDINFKBot/Handlers/MessageHandler.cs:304]: Use the SocketMessage object

File: ETHDINFKBot/Handlers/MessageHandler.cs [hunk 2]
```diff
@@ -297,7 +299,14 @@ private async void EmojiDetection()
...
+                                    } 
+                                    else
+                                    {
+                                        await SocketMessage.Channel.SendFileAsync(stream, $"{emote.EmoteName}.gif");
```
Comment (BattleRush) [inline ETHDINFKBot/Handlers/MessageHandler.cs:308]: As Discord.MessageReference can accept null you could do

`await SocketMessage.Channel.SendFileAsync(stream, $"{emote.EmoteName}.gif", "", false, null, null, false, null, new Discord.MessageReference(SocketMessage.ReferencedMessage?.Id));`

and remove the if statement to simplify code

File: ETHDINFKBot/Handlers/MessageHandler.cs [hunk 3]
```diff
@@ -310,7 +319,14 @@ private async void EmojiDetection()
                                 var resImage = CommonHelper.ResizeImage(bmp, Math.Min(bmp.Height, 64));
                                 var stream = CommonHelper.GetStream(resImage);
 
-                                await SocketMessage.Channel.SendFileAsync(stream, $"{emote.EmoteName}.png");
+                                if (Context.Message.ReferencedMessage != null)
```
Comment (BattleRush) [inline ETHDINFKBot/Handlers/MessageHandler.cs:322]: Similar to above with .gif emotes

**Review by BattleRush — CHANGES_REQUESTED** [review]

**Comment by 4ndriH** [comment]
like dis?

**Pull request merged**